src/backend/joanie/core/tasks/payment_schedule.py

```diff
@@ -27,14 +26,12 @@ def process_today_installment(order_id):
             and installment["state"] == enums.PAYMENT_STATE_PENDING
         ):
             payment_backend = get_payment_backend()
-            try:
-                credit_card = CreditCard.objects.get(owner=order.owner, is_main=True)
-            except CreditCard.DoesNotExist:
+            if not order.credit_card or not order.credit_card.token:
```

Comment:
That means we must save a token for a payment when the user is paying a previous refused installment.
IMO we should replace `ASK_REGISTER_PAY` by `REGISTER_PAY` in the `create_payment` backend method of Lyra

Reply:
I'm not sure, as the user may manually pay a failed installment with a card he don't want to save.
I added a TODO anyway.

Reply:
I'm not confident with this change. We are adding an endpoint to add a tokenized card to a user account and this card will not be linked to an order. We should still check if a card is available, there is something more clever to do IMO.
Add a todo to improve this part